docs/DeveloperGuide.md

```diff
@@ -318,6 +318,44 @@ The following activity diagram summarizes what happens when a user executes a ne
   * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
   * Cons: We must ensure that the implementation of each individual command are correct.
 
+
+### Setting a Default Group for a particular Mod
+
+#### Aim of the feature
+This feature helps facilitate a better user experience and aims to follow through on TAilor's primary objective of
+making administrative tasks less tedious and rudimentary. By setting a default group, the use need not worry
+about repeatedly entering the same group value for several students over an extended period of time.
+
+#### Implementation
+The following classes were created in the process of implementing the `set-default-group` command:
+
+Logic:
+* SetDefaultGroupCommand (and its parser)
+
+Model:
+* ModuleList
+* UniqueModuleList
+* DuplicateModuleException
+* ModuleNotFoundException
+
+The core idea behind this implementation is that every Mod object has a `defaultGroup` attribute that initially is
+unassigned. Once the user enters the command `set-default-group m/MOD g/GROUP`, the `defaultGroup` value for
+that "MOD" gets set to "GROUP". If the command is entered again, the value gets updated and the user is notified.
+
+The sequence diagram for the command `set-default-group m/CS2103T g/W12-1` follows:
+<img src="images/SetDefaultSequenceDiagram.png" width="550"/>
+
+#### Design Considerations
+
+**Aspect: Extendibility**
```

Comment:
I'm not too sure if "Extendibility" is the right word to use here, as it implies more of how the code can be extended or built upon from your contributions.

A better would maybe would be "Conformity", like how your contributions merge together with the code base, or "Accessibility", like how your contributions make the usage of the app more accessible.

Reply:
I see your point. Let me rename it:)